### PR TITLE
Tim i08

### PR DESCRIPTION
**Pull request by Timofey-Vorobev**
Implemented

### CODE REVIEW

File: php/Utils/Strings.php [hunk 1]
```diff
@@ -30,7 +42,16 @@ public function kebabCase(): string
      */
     public function countCharOccurence(string $char): int
     {
-        // @TODO: implement body and cover with tests
+        if (!is_string($char) || strlen($char) != 1) {
```
Comment (p5ych0) [inline php/Utils/Strings.php]: first check is not valid since TypeError introduced and thrown

File: php/Utils/Strings.php [hunk 2]
```diff
@@ -30,7 +42,16 @@ public function kebabCase(): string
...
+            throw new \InvalidArgumentException('Passed value is not a character');
+        }
+        $count = 0;
+        for ($i = 0; $i < strlen($this->string); $i++) {
+            if (strtolower($this->string[$i]) == strtolower($char)) {
```
Comment (p5ych0) [inline php/Utils/Strings.php]: too many computations

File: php/Utils/Strings.php [hunk 3]
```diff
@@ -30,7 +42,16 @@ public function kebabCase(): string
      */
     public function countCharOccurence(string $char): int
```
Comment (p5ych0) [inline php/Utils/Strings.php]: we live in the modern world and so have to support multibyte alphabets

File: php/Utils/Strings.php [hunk 4]
```diff
@@ -9,6 +9,18 @@ class Strings
      */
     protected $string;
 
+    /**
+     * Assigns the supplied value to internal $string field
+     * @param string $data
+     */
+    public function __construct(string $data)
```
Comment (p5ych0) [inline php/Utils/Strings.php]: imagine there's already a constructor which looks like `__construct(BladyInterface $int)`

File: php/Utils/Strings.php [hunk 5]
```diff
@@ -9,6 +9,16 @@ class Strings
      */
     protected $string;
 
+    /**
+     * Assigns the supplied value to internal $string field
+     * @param string $data
+     */
+    public function setValue(string $data)
```
Comment (p5ych0) [inline php/Utils/Strings.php]: Mock object, the reset is fine

**Comment by p5ych0** [comment]
FYI CircleCI fails so far, don't bother

**Comment by Timofey-Vorobev** [comment]
changed implementation to mb_string functions

**Pull request closed without merging**